keyboards/k_type/rules.mk

```diff
@@ -11,21 +11,27 @@ BOARD = IC_TEENSY_3_1
 BOOTLOADER = kiibohd
 
 # Build Options
-#   comment out to disable the options.
+#   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = no	# Virtual DIP switch configuration
-## (Note that for BOOTMAGIC on Teensy LC you have to use a custom .ld script.)
-MOUSEKEY_ENABLE = yes	# Mouse keys
-EXTRAKEY_ENABLE = yes	# Audio control and System control
-CONSOLE_ENABLE = yes	# Console for debug
-COMMAND_ENABLE = yes    # Commands for debug and configuration
-SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
-NKRO_ENABLE = yes	    # USB Nkey Rollover
-CUSTOM_MATRIX = yes # Custom matrix file
-DEBUG_ENABLE = yes
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+CUSTOM_MATRIX = lite
```

Comment:
Also, it looks like may not need a custom matrix? 
Or is it handling different enough?

Reply:
Some of the ARM pin definitions conflict with some of the Kinetis stuff in ChibiOS: https://github.com/qmk/ChibiOS-Contrib/blob/HEAD/os/common/ext/CMSIS/KINETIS/k20xx.h#L105

We could maybe have a special set of pin definitions for the Kinetis (the datasheet seems to prefix them with `PT` instead of just `P`), but for now, just have to use custom matrix.

Reply:
Have you tried like <https://github.com/qmk/qmk_firmware/blob/master/keyboards/handwired/onekey/teensy_32/config.h#L19>?

Reply:
Oh, that's trippy. It complains about B18 and B19 not being defined, adding them into config_common.h (but not including it!) makes it compile without custom matrix.